src/OrchardCore.Modules/OrchardCore.UrlRewriting/Drivers/UrlRewriteRuleDisplayDriver.cs

```diff
@@ -58,14 +59,18 @@ await context.Updater.TryUpdateModelAsync(model, Prefix,
 
         if (string.IsNullOrWhiteSpace(model.SubstitutionUrl))
         {
-            context.Updater.ModelState.AddModelError(Prefix, nameof(model.SubstitutionUrl), S["The rewrite URL is required"]);
+            context.Updater.ModelState.AddModelError(Prefix, nameof(model.SubstitutionUrl), S["The Rewrite URL is required"]);
+        }
+        else if (!Uri.TryCreate(model.SubstitutionUrl, UriKind.RelativeOrAbsolute, out var _))
```

Comment:
SubstitutionUrl can't be validated with Uri as it can contain pattern match chunks like in example: `RewriteRule /img/(.*) /media/$1`  so it is Url + ECMA-262 RegExp

Reply:
Then it should not be called URL. Maybe SubstitutionPattern?